### PR TITLE
Documemnt how to terminate SSL at the ingress LB for WebLogic console access

### PR DESCRIPTION
**Pull request by anpanigr**
Refer to JIRA
  https://jira.oraclecorp.com/jira/browse/OWLS-83840

Also modified the apache values.yaml file to de-reference to  store/oracle/apache:12.2.1.3

### CODE REVIEW

File: kubernetes/samples/charts/traefik/README.md [hunk 1]
```diff
@@ -91,6 +91,80 @@ TLS_PORT=`kubectl -n traefik get service traefik-operator -o jsonpath='{.spec.po
 $ curl -k -H 'host: domain1.org' https://${HOSTNAME}:${TLS_PORT}/testwebapp/
 ```
 
+## SSL Termination at Ingress Controller
```
Comment (rosemarymarano) [inline kubernetes/samples/charts/traefik/README.md:94]: SSL Termination at Ingress Controller -> SSL termination at ingress controller (we use sentence capitalization)

File: kubernetes/samples/charts/traefik/README.md [hunk 2]
```diff
@@ -91,6 +91,80 @@ TLS_PORT=`kubectl -n traefik get service traefik-operator -o jsonpath='{.spec.po
...
+This sample demonstrates how to terminate SSL trafik at ingress contoller to access WebLogic console thru SSL port.
```
Comment (rosemarymarano) [inline kubernetes/samples/charts/traefik/README.md:95]: trafik-> Is this word supposed to be traffic or Traefik?
contoller (typo) -> controller
access WebLogic console -> access the WebLogic Server Administration Console (this is a legal product name) 
thru (not a word) -> through the

File: kubernetes/samples/charts/traefik/README.md [hunk 3]
```diff
@@ -91,6 +91,80 @@ TLS_PORT=`kubectl -n traefik get service traefik-operator -o jsonpath='{.spec.po
 $ curl -k -H 'host: domain1.org' https://${HOSTNAME}:${TLS_PORT}/testwebapp/
 ```
 
+## SSL Termination at Ingress Controller
+This sample demonstrates how to terminate SSL trafik at ingress contoller to access WebLogic console thru SSL port. 
+
+### 1. Enable "WebLogic Plugin Enabled" on WebLogic Domain level
```
Comment (rosemarymarano) [inline kubernetes/samples/charts/traefik/README.md:97]: on WebLogic Domain level -> on the WebLogic domain.

File: kubernetes/samples/charts/traefik/README.md [hunk 4]
```diff
@@ -91,6 +91,80 @@ TLS_PORT=`kubectl -n traefik get service traefik-operator -o jsonpath='{.spec.po
...
+
+If you are using wdt to configure WebLogic Domain, you need to add the following resource section at the domain level to model yaml file.
```
Comment (rosemarymarano) [inline kubernetes/samples/charts/traefik/README.md:99]: wdt  -> WDT
Domain -> domain
yaml  -> YAML

File: kubernetes/samples/charts/traefik/README.md [hunk 5]
```diff
@@ -91,6 +91,80 @@ TLS_PORT=`kubectl -n traefik get service traefik-operator -o jsonpath='{.spec.po
...
+```
+resources:
+     WebAppContainer:
+         WeblogicPluginEnabled: true
+```
+If you are using WLST script to configure domain, following modification needed to respective py script.
```
Comment (rosemarymarano) [inline kubernetes/samples/charts/traefik/README.md:105]: If you are using WLST script to configure domain, following modification needed to respective py script.-> If you are using a WLST script to configure the domain, then the following modifications are needed to the respective PY script.

File: kubernetes/samples/charts/traefik/README.md [hunk 6]
```diff
@@ -91,6 +91,80 @@ TLS_PORT=`kubectl -n traefik get service traefik-operator -o jsonpath='{.spec.po
...
+cd('/Servers/AdminServer')
+set('ListenPort', admin_port)
+set('Name', admin_server_name)
+set('WeblogicPluginEnabled',true)
+...
+cd('/Clusters/%s' % cluster_name)
+set('WeblogicPluginEnabled',true)
+```
+### 2. Update the Ingress resource with customRequestHeaders value.
+Replace the string 'weblogic-domain' with namespace of the WebLogic domain, the string 'domain1' with domain UID and the string 'adminserever' with name of the Admin server in the WebLogic domain.
```
Comment (rosemarymarano) [inline kubernetes/samples/charts/traefik/README.md:117]: adminserever (typo) -> adminserver
Admin server -> Administration Server

File: kubernetes/samples/charts/traefik/README.md [hunk 7]
```diff
@@ -91,6 +91,80 @@ TLS_PORT=`kubectl -n traefik get service traefik-operator -o jsonpath='{.spec.po
...
+      Wl-Proxy-Ssl: "true"
+    sslRedirect: true
+```
+### 3. Create Ingress resource.
+Save the above configuration as 'traefik-tls-console.yaml'.
+```
+ kubectl create -f traefik-tls-console.yaml
+```
+### 4. Access WebLogic console using https port
+Get the SSL port from the kubernates service
```
Comment (rosemarymarano) [inline kubernetes/samples/charts/traefik/README.md:161]: kubernates (typo) -> Kubernetes

File: kubernetes/samples/charts/voyager/README.md [hunk 8]
```diff
@@ -118,6 +118,66 @@ $ curl -k -H 'host: domain1.org' https://${HOSTNAME}:30305/testwebapp/
 $ curl -k -H 'host: domain2.org' https://${HOSTNAME}:30307/testwebapp/
 ```
 
+## SSL Termination at Ingress Controller
```
Comment (rosemarymarano) [inline kubernetes/samples/charts/voyager/README.md:121]: Please make the same corrections here as the ones above for the Traefik README.

File: kubernetes/samples/charts/traefik/README.md (hunk 1 above)
Comment (anpanigr) [reply kubernetes/samples/charts/traefik/README.md]: done

File: kubernetes/samples/charts/traefik/README.md (hunk 2 above)
Comment (anpanigr) [reply kubernetes/samples/charts/traefik/README.md]: done

File: kubernetes/samples/charts/traefik/README.md (hunk 3 above)
Comment (anpanigr) [reply kubernetes/samples/charts/traefik/README.md]: done

File: kubernetes/samples/charts/traefik/README.md (hunk 4 above)
Comment (anpanigr) [reply kubernetes/samples/charts/traefik/README.md]: done

File: kubernetes/samples/charts/traefik/README.md (hunk 5 above)
Comment (anpanigr) [reply kubernetes/samples/charts/traefik/README.md]: done

File: kubernetes/samples/charts/traefik/README.md (hunk 6 above)
Comment (anpanigr) [reply kubernetes/samples/charts/traefik/README.md]: done

File: kubernetes/samples/charts/traefik/README.md (hunk 7 above)
Comment (anpanigr) [reply kubernetes/samples/charts/traefik/README.md]: done

File: kubernetes/samples/charts/voyager/README.md (hunk 8 above)
Comment (anpanigr) [reply kubernetes/samples/charts/voyager/README.md]: done

File: kubernetes/samples/charts/traefik/README.md [hunk 9]
```diff
@@ -91,18 +91,18 @@ TLS_PORT=`kubectl -n traefik get service traefik-operator -o jsonpath='{.spec.po
 $ curl -k -H 'host: domain1.org' https://${HOSTNAME}:${TLS_PORT}/testwebapp/
 ```
 
-## SSL Termination at Ingress Controller
-This sample demonstrates how to terminate SSL trafik at ingress contoller to access WebLogic console thru SSL port. 
+## SSL termination at ingress controller
+This sample demonstrates how to terminate SSL traffic at ingress  to access the WebLogic Server Administration Console through SSL port.
```
Comment (rosemarymarano) [inline kubernetes/samples/charts/traefik/README.md:95]: at ingress  to access -> ??? at the ingress controller to access
through SSL port -> through the SSL port

File: kubernetes/samples/charts/traefik/README.md [hunk 10]
```diff
@@ -91,18 +91,18 @@ TLS_PORT=`kubectl -n traefik get service traefik-operator -o jsonpath='{.spec.po
...
-## SSL Termination at Ingress Controller
-This sample demonstrates how to terminate SSL trafik at ingress contoller to access WebLogic console thru SSL port. 
+## SSL termination at ingress controller
+This sample demonstrates how to terminate SSL traffic at ingress  to access the WebLogic Server Administration Console through SSL port. 
 
-### 1. Enable "WebLogic Plugin Enabled" on WebLogic Domain level
+### 1. Enable "WebLogic Plugin Enabled" on the WebLogic domain level
 
-If you are using wdt to configure WebLogic Domain, you need to add the following resource section at the domain level to model yaml file.
+If you are using WDT to configure the WebLogic domain, you need to add the following resource section at the domain level to model YAML file.
```
Comment (rosemarymarano) [inline kubernetes/samples/charts/traefik/README.md:99]: to model -> to the model

File: kubernetes/samples/charts/traefik/README.md [hunk 11]
```diff
@@ -91,18 +91,18 @@ TLS_PORT=`kubectl -n traefik get service traefik-operator -o jsonpath='{.spec.po
...
 ```
 resources:
      WebAppContainer:
          WeblogicPluginEnabled: true
 ```
-If you are using WLST script to configure domain, following modification needed to respective py script.
+If you are using WLST script to configure the domain, then the following modifications are needed to the respective PY script.
```
Comment (rosemarymarano) [inline kubernetes/samples/charts/traefik/README.md:105]: using WLST script -> using a WLST script

File: kubernetes/samples/charts/traefik/README.md [hunk 12]
```diff
@@ -157,13 +157,13 @@ Save the above configuration as 'traefik-tls-console.yaml'.
 ```
  kubectl create -f traefik-tls-console.yaml
 ```
-### 4. Access WebLogic console using https port
-Get the SSL port from the kubernates service 
+### 4. Access the WebLogic Server Administration Console using https port
```
Comment (rosemarymarano) [inline kubernetes/samples/charts/traefik/README.md:160]: using https port-> using the HTTPS port

File: kubernetes/samples/charts/voyager/README.md [hunk 13]
```diff
@@ -118,18 +118,18 @@ $ curl -k -H 'host: domain1.org' https://${HOSTNAME}:30305/testwebapp/
 $ curl -k -H 'host: domain2.org' https://${HOSTNAME}:30307/testwebapp/
 ```
 
-## SSL Termination at Ingress Controller
-This sample demonstrates how to terminate SSL trafik at ingress contoller to access WebLogic console thru SSL port. 
+## SSL termination at ingress controller
+This sample demonstrates how to terminate SSL traffic at ingress controller to access the WebLogic Server Administration Console through SSL port.
```
Comment (rosemarymarano) [inline kubernetes/samples/charts/voyager/README.md:122]: at ingress controller -> at the ingress controller
through SSL port-> through the SSL port

File: kubernetes/samples/charts/voyager/README.md [hunk 14]
```diff
@@ -118,18 +118,18 @@ $ curl -k -H 'host: domain1.org' https://${HOSTNAME}:30305/testwebapp/
...
-## SSL Termination at Ingress Controller
-This sample demonstrates how to terminate SSL trafik at ingress contoller to access WebLogic console thru SSL port. 
+## SSL termination at ingress controller
+This sample demonstrates how to terminate SSL traffic at ingress controller to access the WebLogic Server Administration Console through SSL port. 
 
-### 1. Enable "WebLogic Plugin Enabled" on WebLogic Domain level
+### 1. Enable "WebLogic Plugin Enabled" on the WebLogic domain level
 
-If you are using wdt to configure WebLogic Domain, you need to add the following resource section at the domain level to model yaml file.
+If you are using WDT to configure the WebLogic domain, you need to add the following resource section at the domain level to model YAML file.
```
Comment (rosemarymarano) [inline kubernetes/samples/charts/voyager/README.md:126]: to model -> to the model

File: kubernetes/samples/charts/voyager/README.md [hunk 15]
```diff
@@ -118,18 +118,18 @@ $ curl -k -H 'host: domain1.org' https://${HOSTNAME}:30305/testwebapp/
...
 ```
 resources:
      WebAppContainer:
          WeblogicPluginEnabled: true
 ```
-If you are using WLST script to configure domain, following modification needed to respective py script.
+If you are using WLST script to configure the domain, then the following modificationis are needed to respective PY script.
```
Comment (rosemarymarano) [inline kubernetes/samples/charts/voyager/README.md:132]: using WLST script -> using a WLST script
modificationis (typo) -> modifications
to respective -> to the respective

File: kubernetes/samples/charts/voyager/README.md [hunk 16]
```diff
@@ -141,7 +141,7 @@ cd('/Clusters/%s' % cluster_name)
 set('WeblogicPluginEnabled',true)
 ```
 ### 2. Update the frontendRules section Ingress resource.
-Replace the string 'weblogic-domain' with namespace of the WebLogic domain, the string 'domain1' with domain UID and the string 'adminserever' with name of the Admin server in the WebLogic domain.
+Replace the string 'weblogic-domain' with namespace of the WebLogic domain, the string 'domain1' with domain UID and the string 'adminserver' with name of the Administration server in the WebLogic domain.
```
Comment (rosemarymarano) [inline kubernetes/samples/charts/voyager/README.md:144]: Administration server -> Administration Server

File: kubernetes/samples/charts/voyager/README.md [hunk 17]
```diff
@@ -175,8 +175,8 @@ Save the above configuration as 'voyager-tls-console.yaml'.
 ```
  kubectl create -f voyager-tls-console.yaml
 ```
-### 4. Access WebLogic console using https port
-In a web browser type 'https://${HOSTNAME}:30443/console' in the address bar to access the console. 
+### 4. Access the WebLogic Server Administration Console using https port
```
Comment (rosemarymarano) [inline kubernetes/samples/charts/voyager/README.md:178]: using https port -> using the HTTPS port

File: kubernetes/samples/charts/traefik/README.md (hunk 9 above)
Comment (anpanigr) [reply kubernetes/samples/charts/traefik/README.md]: done

File: kubernetes/samples/charts/traefik/README.md (hunk 10 above)
Comment (anpanigr) [reply kubernetes/samples/charts/traefik/README.md]: done

File: kubernetes/samples/charts/traefik/README.md (hunk 11 above)
Comment (anpanigr) [reply kubernetes/samples/charts/traefik/README.md]: done

File: kubernetes/samples/charts/traefik/README.md (hunk 12 above)
Comment (anpanigr) [reply kubernetes/samples/charts/traefik/README.md]: done

File: kubernetes/samples/charts/voyager/README.md (hunk 13 above)
Comment (anpanigr) [reply kubernetes/samples/charts/voyager/README.md]: done

File: kubernetes/samples/charts/voyager/README.md (hunk 14 above)
Comment (anpanigr) [reply kubernetes/samples/charts/voyager/README.md]: done

File: kubernetes/samples/charts/voyager/README.md (hunk 16 above)
Comment (anpanigr) [reply kubernetes/samples/charts/voyager/README.md]: done

File: kubernetes/samples/charts/voyager/README.md (hunk 17 above)
Comment (anpanigr) [reply kubernetes/samples/charts/voyager/README.md]: done

File: kubernetes/samples/charts/voyager/README.md (hunk 15 above)
Comment (anpanigr) [reply kubernetes/samples/charts/voyager/README.md]: done

File: kubernetes/samples/charts/voyager/README.md [hunk 18]
```diff
@@ -21,7 +21,7 @@ $ helm search repo appscode/voyager
 NAME               CHART VERSION APP VERSION	DESCRIPTION
 appscode/voyager   v12.0.0       v12.0.0    	Voyager by AppsCode - Secure HAProxy Ingress Co...
 ```
-> **NOTE**: After updating the helm repository, the Voyager version listed maybe newer that the one appearing here, please check with the Voyager site for the lastest supported versions.
+> **NOTE**: After updating the helm repository, the Voyager version listed maybe newer that the one appearing here, please check with the Voyager site for the latest supported versions.
```
Comment (rosemarymarano) [inline kubernetes/samples/charts/voyager/README.md:24]: helm  -> Helm (capitalize)
maybe  -> may be (two words)
here, please -> here. Please (two separate sentences)

File: kubernetes/samples/charts/voyager/README.md [hunk 19]
```diff
@@ -21,7 +21,7 @@ $ helm search repo appscode/voyager
...
 
 ### 2. Install the Voyager operator
```
Comment (rosemarymarano) [inline kubernetes/samples/charts/voyager/README.md:26]: Wait until the Voyager Operator is running. -> Wait until the Voyager operator is running. (You do not capitalize operator elsewhere, don't do it here on line 38.)
Please change ALL instances of "Ingress" (capital I) to ingress (lower case).

File: kubernetes/samples/charts/voyager/README.md [hunk 20]
```diff
@@ -118,6 +118,64 @@ $ curl -k -H 'host: domain1.org' https://${HOSTNAME}:30305/testwebapp/
...
+If you are using a WLST script to configure the domain, then the following modifications are needed to the respective PY script.
+```
+# Configure the Administration Server
+cd('/Servers/AdminServer')
+set('WeblogicPluginEnabled',true)
+...
+cd('/Clusters/%s' % cluster_name)
+set('WeblogicPluginEnabled',true)
+```
+### 2. Update the frontendRules section Ingress resource.
```
Comment (rosemarymarano) [inline kubernetes/samples/charts/voyager/README.md:141]: Please make ALL the same edits to this section of the doc, that I gave you for the same section in Traefik README. I do not want to re-print them all here but please make sure that you incorporate them all.

File: kubernetes/samples/charts/traefik/README.md [hunk 21]
```diff
@@ -91,6 +91,78 @@ TLS_PORT=`kubectl -n traefik get service traefik-operator -o jsonpath='{.spec.po
...
+metadata:
+  name: tls-console-middleware
+  namespace: weblogic-domain
+spec:
+  headers:
+    customRequestHeaders:
+      Wl-Proxy-Ssl: "true"
+    sslRedirect: true
+```
+### 3. Create Ingress resource.
```
[the remark this answers is not in the document]
Comment (anpanigr) [reply kubernetes/samples/charts/traefik/README.md]: done

File: kubernetes/samples/charts/traefik/README.md [hunk 22]
```diff
@@ -91,6 +91,78 @@ TLS_PORT=`kubectl -n traefik get service traefik-operator -o jsonpath='{.spec.po
...
+Save the above configuration as 'traefik-tls-console.yaml'.
+```
+ kubectl create -f traefik-tls-console.yaml
+```
+### 4. Access the WebLogic Server Administration Console using the HTTPS port
+Get the SSL port from the Kubernetes service
```
[the remark this answers is not in the document]
Comment (anpanigr) [reply kubernetes/samples/charts/traefik/README.md]: done

File: kubernetes/samples/charts/traefik/README.md [hunk 23]
```diff
@@ -91,6 +91,78 @@ TLS_PORT=`kubectl -n traefik get service traefik-operator -o jsonpath='{.spec.po
...
+```
+ kubectl create -f traefik-tls-console.yaml
+```
+### 4. Access the WebLogic Server Administration Console using the HTTPS port
+Get the SSL port from the Kubernetes service 
+```
+# Get the ingress controller secure web port
+SSLPORT=$(kubectl -n traefik get service traefik-operator -o jsonpath='{.spec.ports[?(@.name=="websecure")].nodePort}')
+```
+In a web browser type 'https://${HOSTNAME}:${SSLPORT}/console' in the address bar to access the WebLogic Server Administration Console.
```
[the remark this answers is not in the document]
Comment (anpanigr) [reply kubernetes/samples/charts/traefik/README.md]: done

File: kubernetes/samples/charts/voyager/README.md (hunk 18 above)
Comment (anpanigr) [reply kubernetes/samples/charts/voyager/README.md]: done

File: kubernetes/samples/charts/voyager/README.md (hunk 19 above)
Comment (anpanigr) [reply kubernetes/samples/charts/voyager/README.md:26]: done

File: kubernetes/samples/charts/voyager/README.md (hunk 20 above)
Comment (anpanigr) [reply kubernetes/samples/charts/voyager/README.md]: Also modified the traefik README.md

File: kubernetes/samples/charts/traefik/README.md [hunk 24]
```diff
@@ -91,6 +91,79 @@ TLS_PORT=`kubectl -n traefik get service traefik-operator -o jsonpath='{.spec.po
...
+# Configure the Administration Server
+cd('/Servers/AdminServer')
+set('WeblogicPluginEnabled',true)
+...
+cd('/Clusters/%s' % cluster_name)
+set('WeblogicPluginEnabled',true)
+```
+### 2. Update the ingress resource with customRequestHeaders value
+Replace the string `weblogic-domain` with namespace of the WebLogic domain, the string `domain1` with domain UID and the string `adminserver` with name of the Administration Server in the WebLogic domain.  
+
```
Comment (mriccell) [inline kubernetes/samples/charts/traefik/README.md:116]: Add the following note
NOTE: If you also have HTTP coming into ingress you will want to make sure that you remove any incoming WL-Proxy-SSL header. This protects you from a malicious user sending in a request and tricking WebLogic into thinking it’s secure when it wasn’t.  Add the following rules in the Traefik proxy configuration to block any WL-Proxy headers coming from the client.

File: kubernetes/samples/charts/nginx/README.md [hunk 25]
```diff
@@ -77,6 +77,23 @@ $ export LB_PORT=$(kubectl -n nginx get service nginx-operator-ingress-nginx-con
...
+
+```
+# Get the ingress controller web port
+$ export LB_PORT=$(kubectl -n nginx get service nginx-operator-ingress-nginx-controller -o jsonpath='{.spec.ports[?(@.name=="http")].nodePort}')
+$ curl http://${HOSTNAME}:${LB_PORT}/domain1/testwebapp/
+$ curl http://${HOSTNAME}:${LB_PORT}/domain2/testwebapp/
+```
+
 #### Host-based secured routing
 This sample demonstrates how to access an application on two WebLogic domains using an HTTPS endpoint. Install a TLS-enabled ingress controller.
```
Comment (mriccell) [inline kubernetes/samples/charts/nginx/README.md:98]: Add the following note
NOTE: If you also have HTTP coming into ingress you will want to make sure that you remove any incoming WL-Proxy-SSL header. This protects you from a malicious user sending in a request and tricking WebLogic into thinking it’s secure when it wasn’t.  Add the following rules in the Nginx proxy configuration to block any WL-Proxy headers coming from the client.

File: kubernetes/samples/charts/voyager/README.md [hunk 26]
```diff
@@ -118,8 +118,66 @@ $ curl -k -H 'host: domain1.org' https://${HOSTNAME}:30305/testwebapp/
...
+# Configure the Administration Server
+cd('/Servers/AdminServer')
+set('WeblogicPluginEnabled',true)
+...
+cd('/Clusters/%s' % cluster_name)
+set('WeblogicPluginEnabled',true)
+```
+### 2. Update the frontendRules section ingress resource
+Replace the string `weblogic-domain` with namespace of the WebLogic domain, the string `domain1` with domain UID and the string `adminserver` with name of the Administration Server in the WebLogic domain.
+```
```
Comment (mriccell) [inline kubernetes/samples/charts/voyager/README.md:143]: After finding how to strip the header from Voyager please add the following note
NOTE: If you also have HTTP coming into ingress you will want to make sure that you remove any incoming WL-Proxy-SSL header. This protects you from a malicious user sending in a request and tricking WebLogic into thinking it’s secure when it wasn’t.  Add the following rules in the Voyager proxy configuration to block any WL-Proxy headers coming from the client.

**Review by mriccell — CHANGES_REQUESTED** [review]

File: kubernetes/samples/charts/traefik/README.md [hunk 27]
```diff
@@ -91,6 +91,78 @@ TLS_PORT=`kubectl -n traefik get service traefik-operator -o jsonpath='{.spec.po
...
+If you are using a WLST script to configure the domain, then the following modifications are needed to the respective PY script.
+```
+# Configure the Administration Server
+cd('/Servers/AdminServer')
+set('WeblogicPluginEnabled',true)
+...
+cd('/Clusters/%s' % cluster_name)
+set('WeblogicPluginEnabled',true)
+```
+### 2. Update the Ingress resource with customRequestHeaders value.
```
[the remark this answers is not in the document]
Comment (anpanigr) [reply kubernetes/samples/charts/traefik/README.md]: done

File: kubernetes/samples/charts/traefik/README.md (hunk 24 above)
Comment (anpanigr) [reply kubernetes/samples/charts/traefik/README.md:116]: done

File: kubernetes/samples/charts/nginx/README.md (hunk 25 above)
Comment (anpanigr) [reply kubernetes/samples/charts/nginx/README.md:98]: done

File: kubernetes/samples/charts/nginx/README.md [hunk 28]
```diff
@@ -124,7 +141,9 @@ cd('/Clusters/%s' % cluster_name)
 set('WeblogicPluginEnabled',true)
 ```
 ### 2. Create NGINX ingress resource with custom annotation values
-Save the below configuration as 'nginx-tls-console.yaml' and replace the string 'weblogic-domain' with the namespace of the WebLogic domain, the string 'domain1' with the domain UID, and the string 'adminserver' with the name of the Administration server in the WebLogic domain.
+Save the below configuration as `nginx-tls-console.yaml` and replace the string `weblogic-domain` with the namespace of the WebLogic domain, the string `domain1` with the domain UID, and the string `adminserver` with the name of the Administration Server in the WebLogic domain.
+
+**NOTE**: If you also have HTTP coming into ingress you will want to make sure that you remove any incoming WL-Proxy-SSL header. This protects you from a malicious user sending in a request and tricking WebLogic into thinking it's secure when it wasn't. Add the following annotations in the Nginx ingress configuration to block any WL-Proxy headers coming from the client. In the following example the ingress resource will eliminate client header `WL-Proxy-Client-IP`.
```
Comment (rosemarymarano) [inline kubernetes/samples/charts/nginx/README.md:146]: This NOTE needs to be re-written as follows: 
**NOTE**: If you also have HTTP requests coming into an ingress, make sure that you remove any incoming `WL-Proxy-SSL` header. This protects you from a malicious user sending in a request to appear to WebLogic as secure when it isn't. Add the following annotations in the NGINX ingress configuration to block `WL-Proxy` headers coming from the client. In the following example, the ingress resource will eliminate the client header `WL-Proxy-Client-IP`.

File: kubernetes/samples/charts/traefik/README.md [hunk 29]
```diff
@@ -91,6 +91,83 @@ TLS_PORT=`kubectl -n traefik get service traefik-operator -o jsonpath='{.spec.po
...
+cd('/Servers/AdminServer')
+set('WeblogicPluginEnabled',true)
+...
+cd('/Clusters/%s' % cluster_name)
+set('WeblogicPluginEnabled',true)
+```
+### 2. Update the ingress resource with customRequestHeaders value
+Replace the string `weblogic-domain` with namespace of the WebLogic domain, the string `domain1` with domain UID and the string `adminserver` with name of the Administration Server in the WebLogic domain.  
+
+**NOTE**: If you also have HTTP coming into ingress you will want to make sure that you remove any incoming WL-Proxy-SSL header. This protects you from a malicious user sending in a request and tricking WebLogic into thinking it's secure when it wasn't. Add the following `customRequestHeaders` in the Traefik ingress configuration to block any WL-Proxy headers coming from the client. In the following example the ingress resource will eliminate client header `WL-Proxy-Client-IP`.
```
Comment (rosemarymarano) [inline kubernetes/samples/charts/traefik/README.md:117]: Edit the text of this NOTE as follows:
**NOTE**: If you also have HTTP requests coming into an ingress, make sure that you remove any incoming `WL-Proxy-SSL` header. This protects you from a malicious user sending in a request to appear to WebLogic as secure when it isn't. Add the following `customRequestHeaders` in the Traefik ingress configuration to block `WL-Proxy` headers coming from the client. In the following example, the ingress resource will eliminate the client header `WL-Proxy-Client-IP`.

File: kubernetes/samples/charts/voyager/README.md [hunk 30]
```diff
@@ -118,8 +118,69 @@ $ curl -k -H 'host: domain1.org' https://${HOSTNAME}:30305/testwebapp/
...
+cd('/Servers/AdminServer')
+set('WeblogicPluginEnabled',true)
+...
+cd('/Clusters/%s' % cluster_name)
+set('WeblogicPluginEnabled',true)
+```
+### 2. Update the frontendRules section ingress resource
+Replace the string `weblogic-domain` with namespace of the WebLogic domain, the string `domain1` with domain UID and the string `adminserver` with name of the Administration Server in the WebLogic domain.
+
+**NOTE**: If you also have HTTP coming into ingress you will want to make sure that you remove any incoming WL-Proxy-SSL header. This protects you from a malicious user sending in a request and tricking WebLogic into thinking it's secure when it wasn't. Add the following rules in the Voyager ingress configuration to block any WL-Proxy headers coming from the client. In the following example the ingress resource will eliminate client header `WL-Proxy-Client-IP`.
```
Comment (rosemarymarano) [inline kubernetes/samples/charts/voyager/README.md:144]: Please re-write this NOTE as follows:
**NOTE**: If you also have HTTP requests coming into an ingress, make sure that you remove any incoming `WL-Proxy-SSL` header. This protects you from a malicious user sending in a request to appear to WebLogic as secure when it isn't. Add the following rules in the Voyager ingress configuration to block `WL-Proxy` headers coming from the client. In the following example, the ingress resource will eliminate the client header `WL-Proxy-Client-IP`.

**Review by rosemarymarano — APPROVED** [review]
The only remaining edits are in three NOTEs; otherwise, I approve.

**Review by rosemarymarano — APPROVED** [review]

File: kubernetes/samples/charts/nginx/README.md [hunk 31]
```diff
@@ -137,6 +156,7 @@ metadata:
     nginx.ingress.kubernetes.io/configuration-snippet: |
       more_set_input_headers "X-Forwarded-Proto: https";
       more_set_input_headers "WL-Proxy-SSL: true";
+      proxy_set_header WL-Proxy-Client-IP "";
```
Comment (mriccell) [inline kubernetes/samples/charts/nginx/README.md:159]: Is it possible to filter out WL-Proxy-SSL and WL-Proxy-Client-IP?

File: kubernetes/samples/charts/traefik/README.md [hunk 32]
```diff
@@ -91,6 +91,84 @@ TLS_PORT=`kubectl -n traefik get service traefik-operator -o jsonpath='{.spec.po
...
+metadata:
+  name: tls-console-middleware
+  namespace: weblogic-domain
+spec:
+  headers:
+    customRequestHeaders:
+      WL-Proxy-SSL: "true"
+      X-Custom-Request-Header: ""
+      X-Forwarded-For: "" 
+      WL-Proxy-Client-IP: ""
```
Comment (mriccell) [inline kubernetes/samples/charts/traefik/README.md:156]: Is it possible to filter out WL-Proxy-SSL and WL-Proxy-Client-IP?

File: kubernetes/samples/charts/voyager/README.md [hunk 33]
```diff
@@ -118,8 +118,69 @@ $ curl -k -H 'host: domain1.org' https://${HOSTNAME}:30305/testwebapp/
...
+spec:
+  tls:
+  - secretName: domain1-tls-cert
+    hosts: 
+    - '*'
+  frontendRules:
+  - port: 443
+    rules:
+    - http-request set-header WL-Proxy-SSL true
+    - http-request del-header WL-Proxy-Client-IP
```
Comment (mriccell) [inline kubernetes/samples/charts/voyager/README.md:164]: @mriccell


Is it possible to filter out WL-Proxy-SSL and WL-Proxy-Client-IP?

**Review by mriccell — CHANGES_REQUESTED** [review]

File: kubernetes/samples/charts/voyager/README.md (hunk 26 above)
Comment (anpanigr) [reply kubernetes/samples/charts/voyager/README.md:145]: done

frontendRules:
  - port: 443
    rules:
    - http-request del-header WL-Proxy-Client-IP
    - http-request del-header WL-Proxy-SSL
    - http-request set-header WL-Proxy-SSL true

File: kubernetes/samples/charts/voyager/README.md (hunk 33 above)
Comment (anpanigr) [reply kubernetes/samples/charts/voyager/README.md:163]: done

File: kubernetes/samples/charts/traefik/README.md (hunk 32 above)
Comment (anpanigr) [reply kubernetes/samples/charts/traefik/README.md:155]: done

**Review by mriccell — APPROVED** [review]

**Pull request merged**